experimental/packages/opentelemetry-sdk-metrics-base/src/MeterProvider.ts

```diff
@@ -69,9 +120,33 @@ export class MeterProvider implements metrics.MeterProvider {
     this._sharedState.metricCollectors.push(collector);
   }
 
-  addView(view: View, instrumentSelector: InstrumentSelector, meterSelector: MeterSelector) {
-    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view
-    this._sharedState.viewRegistry.addView(view, instrumentSelector, meterSelector);
+  addView(options: ViewOptions) {
+    // the SDK MUST NOT allow Views with a specified name to be declared with instrument selectors that select by instrument type or wildcard
+    if (options.name !== undefined) {
+      if (options.instrument?.type !== undefined) {
+        throw new Error('Views with a specified name must not be declared with instrument selectors that select by instrument type.');
+      }
+      if (options.instrument?.name !== undefined && options.instrument.name.includes('*')) {
```

Comment:
I think this check could be moved to PatternPredicate as a static method so that when we add more kinds of matches we won't lose track of the definition of "wildcard".